recommended edits

AlexRecross opened the pull request:
TabSize = 2 space
Styles divided to personals modules
Re-upload all app on repository
Clean code from overused empty lines etc.

AlexRecross committed:
rePush all app

AlexRecross committed:
styles divided to personals modules

AlexRecross committed:
Merge remote-tracking branch 'origin/courses-app' into courses-app

AlexRecross committed:
authors-map to component

Azimyth commented on src/components/courses/components/course-card/course-card.jsx:
```diff
@@ -0,0 +1,41 @@
+import Button from '../../../../common/button';
+import Duration from "../../../duration/durations";
+
+function CourseCard({courses, authors}) {
```
1) Create separate component CourseCards and move map there
2) Move logic with authors to parent component

AlexRecross replied:
Done!

Azimyth commented on src/components/courses/courses.jsx:
```diff
@@ -0,0 +1,30 @@
+	const [searchValue, setSearchValue] = useState('');
+	const filteredCourses = courses.filter(
+		(course) =>
+			course.title.toLowerCase().includes(searchValue.toLowerCase()) ||
```
useMemo for courses (prepare data for usage, map authors etc)

AlexRecross replied:
need some assistance with this

Azimyth replied:
Not actual

Azimyth commented on src/components/courses/components/search-bar/search-bar.jsx:
```diff
@@ -0,0 +1,31 @@
+
+function SearchBar({ value, onChange, goToCreate }) {
+	return (
+		<div className={styles.searchBar}>
```
It should form tag

AlexRecross replied:
Done

Azimyth commented on src/components/courses/components/course-card/course-card.jsx:
```diff
@@ -0,0 +1,41 @@
+			{courses.map((course) => {
+				return (
+					<li key={course.id} className={styles.courseCard}>
+						<div className={styles.report}>
```
Please use dl - https://developer.mozilla.org/ru/docs/Web/HTML/Element/dl

AlexRecross replied:
Done

AlexRecross committed:
logo in separate component

add new common component 'Input' replace the olds one in progress

<SearchBar> component is form now, and Instead of live search, I made a search by button

Instead of live search, I made a search by button

PROCESSING point

AlexRecross committed:
New common component 'Input' replace an olds one in <createCourse> component

<course-card> in tag <dl> now, with styles

little changes in <searchBar> component

AlexRecross committed:
<create-new-author> turn in new separate component

create <formState> instead of many separates state

processing point

AlexRecross committed:
<authors-map> turn to new separate component, and add all needed functionality. need add memoisation in this component - processing point

need to finish <formState> and add functionality taking into account all changes

AlexRecross committed:
<author-lists> component memoization problem.
processing point
will stop on working version without memoisation

<formState> component problem: on the step setting data outside form(id, authorsIdArray and creationDate) data not field in state on the first try...

processing point...

AlexRecross committed:
<Authors-list /> memoisation component - done

<courses-cards/> component optimisation

<create-course/> component optimisation, still have some bugs

processing point

AlexRecross committed:
all require complete

AlexRecross committed:
Delete all css styles, wrap all App in React-strap styles and components(Input, Form, Button) etc

AlexRecross committed:
add text decoration on authors in course-card, edit code-style in authors lists, some style change in buttons and inputs

AlexRecross committed:
Cards change

Azimyth commented on src/App.jsx:
```diff
@@ -1,37 +1,36 @@
-	const [page, setPage] = useState('courses')
-
+	//common data
 	const [courses, setCourses] = useState(mockedCoursesList)
```
Prepare courses for your needs => map authors id to names,

AlexRecross replied:
Done

Azimyth commented on src/common/to-hours-and-minutes.jsx:
```diff
@@ -0,0 +1,14 @@
+//hours to minutes
+
+export default function toHoursAndMinutes(totalMinutes) {
```
If this func returns just string move it to helpers folder and rename to formatTo....

AlexRecross replied:
Done!

Azimyth commented on src/components/courses/components/course-card/courses-cards.jsx:
```diff
@@ -0,0 +1,42 @@
+function CoursesCards({courses, authors}) {
+
+	function authorsNameById(authorsId) {
+		return authorsId.map(authorId => authors.find(({ id }) => id === authorId)?.name).join(', ')
```
Let's do this formatting on top level

AlexRecross replied:
Done

Azimyth commented on src/components/courses/components/course-card/courses-cards.jsx:
```diff
@@ -0,0 +1,42 @@
+				return (
+					<li key={course.id} className='row border border-success mt-1 mx-0'>
+						<dl className='col-7 m-1' >
+							<dt><b>{course.title}</b></dt>
```
It should not be `dl` in this case, just `h2/h3` and `p`

AlexRecross replied:
Done!

Azimyth commented on src/components/courses/components/course-card/courses-cards.jsx:
```diff
@@ -0,0 +1,42 @@
+								</dd>
+								<dt className='col-4'><b>Created: </b></dt>
+								<dd className='col-8'>{course.creationDate}</dd>
+								<Button color='success' outline>Show course</Button>
```
All except `dt` and `dd `shouldn't be a part of `dl,` so move out Button

AlexRecross replied:
Done!

Azimyth commented on src/components/courses/components/course-card/courses-cards.jsx:
```diff
@@ -0,0 +1,42 @@
+
+	return (
+		<List className='mt-1'>
+			{courses.map((course) => {
```
Use `isEmpty` from lodash.isEmpty and optional chaining in case of operating data to prevent page breaking

Azimyth commented on src/components/create-courses/component/authors-lists.jsx:
```diff
@@ -0,0 +1,71 @@
+
+export default function AuthorsLists({ authorsList, setAuthorsList }) {
+
+	const authorsNotSelected = useMemo( () => {
```
You can remove this memo lists

AlexRecross replied:
Done!

Azimyth commented on src/components/create-courses/component/authors-lists.jsx:
```diff
@@ -0,0 +1,71 @@
+	}, [authorsList])
+
+	function selectToggle(authorId) {
+		const newArray = authorsList.filter((author) => author.id !== authorId);
```
Let's use just map to reduce amount of operations

AlexRecross replied:
Done!

Azimyth commented on src/components/create-courses/component/authors-lists.jsx:
```diff
@@ -0,0 +1,71 @@
+	}
+
+	return <Row xs='1' className='mt-1'>
+		{!authorsNotSelected.length ? <List type="unstyled">
```
Use `isEmpty` from lodash.isEmpty and optional chaining in case of operating data to prevent page breaking

AlexRecross replied:
This memo list Was removed

Azimyth commented on src/components/create-courses/component/authors-lists.jsx:
```diff
@@ -0,0 +1,71 @@
+				<li><b>Available authors empty</b></li>
+			</List> : <List type="unstyled">
+				<b>Available authors:</b>
+				{authorsNotSelected.map((author) => <li  className='row mt-1' key={author.id}>
```
You can operate with authorList from props and use ternary operator to check the render `selected ? <... /> : null`

AlexRecross replied:
Done

Azimyth commented on src/components/create-courses/component/create-new-author.jsx:
```diff
@@ -0,0 +1,58 @@
+	//"Add New Author" functionality
+	const [newAuthorName, setNewAuthorName] = useState('');
+
+	function createNewAuthor(){
```
Move this func outside the component

AlexRecross replied:
i can't remove this func outside componen because  she use hook/state from this component

Azimyth commented on src/components/create-courses/component/create-new-author.jsx:
```diff
@@ -0,0 +1,58 @@
+		return newAuthor
+	}
+
+	const newName = (event) => {
```
Rename to onNameChange or onChangeHandler or something similar

AlexRecross replied:
Done

Azimyth commented on src/components/create-courses/create-course.jsx:
```diff
@@ -0,0 +1,181 @@
+
+	//"Add New Author" functionality
+	function addNewAuthorToList(newAuthor) {
+		newAuthor !== undefined ? addAuthor(newAuthor) : console.log('newAuthor false',newAuthor)
```
Same with check here, isEmpty or just `newAuthor ? .... : ....`

AlexRecross replied:
Done!

Azimyth commented on src/components/create-courses/create-course.jsx:
```diff
@@ -0,0 +1,181 @@
+
+	function getAuthorsId() {
+		const authorsId = []
+		for(let i = 0; i < authorsList.length; i++) {
```
Just simplify this func to reduce

AlexRecross replied:
Done

Azimyth commented on src/components/create-courses/create-course.jsx:
```diff
@@ -0,0 +1,181 @@
+		return authorsId
+	}
+
+	function validateCourseFormData(course) {
```
Move this func ottside component

AlexRecross replied:
Done!

Azimyth commented on src/components/create-courses/create-course.jsx:
```diff
@@ -0,0 +1,181 @@
+
+	function createNewCourse() {
+		const course = formState;
+		course.id= uuidv4();
```
Just try to avoid mutations, the better way is spread operator

`return ({
...formState,
id: uuidv4(),
authors:  authorsList.map(a => ....)
})`

AlexRecross replied:
Done!

Azimyth commented on src/components/create-courses/create-course.jsx:
```diff
@@ -0,0 +1,181 @@
+		const course =createNewCourse()
+		if(!validateCourseFormData(course)){
+			return alert('Form not valid');
+		}else{
```
You can avoid else block statement if you return something in if

AlexRecross replied:
Done

AlexRecross committed:
recommended edits

AlexRecross committed:
final edits

AlexRecross committed:
add isEmpty validator
add some es-lint extends & rules, and edit allApp codeStyle during this change
